src/main/java/com/couchbase/lite/Database.java

```diff
@@ -4064,18 +4075,40 @@ else if (attachInfo.containsKey("follows") && ((Boolean)attachInfo.get("follows"
                 // This means it's already been registered in _pendingAttachmentsByDigest;
                 // I just need to look it up by its "digest" property and install it into the store:
                 installAttachment(attachment, attachInfo);
-
             }
             else {
                 // This item is just a stub; validate and skip it
-                if (((Boolean)attachInfo.get("stub")).booleanValue() == false) {
+                if (((Boolean) attachInfo.get("stub")).booleanValue() == false) {
                     throw new CouchbaseLiteException("Expected this attachment to be a stub", Status.BAD_ATTACHMENT);
                 }
-                int revPos = ((Integer)attachInfo.get("revpos")).intValue();
+                int revPos = ((Integer) attachInfo.get("revpos")).intValue();
                 if (revPos <= 0) {
                     throw new CouchbaseLiteException("Invalid revpos: " + revPos, Status.BAD_ATTACHMENT);
                 }
-                continue;
+                Map<String, Object> parentAttachments = getAttachments(rev.getDocId(), prevRevID);
+                if (parentAttachments != null && parentAttachments.containsKey(name)) {
+                    Map<String, Object> parentAttachment = (Map<String, Object>) parentAttachments.get(name);
+                    try {
+                        BlobKey blobKey = new BlobKey((String) attachInfo.get("digest"));
+                        attachment.setBlobKey(blobKey);
+                    } catch (IllegalArgumentException e) {
+                        continue;
+                    }
+                } else if (parentAttachments == null || !parentAttachments.containsKey(name)) {
+                    BlobKey blobKey = null;
+                    try {
+                        blobKey = new BlobKey((String) attachInfo.get("digest"));
+                    } catch (IllegalArgumentException e) {
+                        continue;
+                    }
+                    if (getAttachments().hasBlobForKey(blobKey)) {
+                        attachment.setBlobKey(blobKey);
+                    } else {
+                        continue;
+                    }
+                } else {
+                    continue;
+                }
             }
```

Comment:
I understand that the code here injects the attachment with the blob key generated by the digest from the attachment info. When the processAttachmentsForRevision method is later called, the copyAttachmentNamedFromSequenceToSequence() method will not get called as the attachment does already exist.

However there is one thing that I'm not so sure is the revpos set in the processAttachmentsForRevision() method. I think it will set the revpos to the current revision instead of the one from the attachment info (Could you be confirm which revpos it will be set?). Honestly I don't know if it is matter or not.